config/config.go

```diff
@@ -635,6 +635,8 @@ type ScrapeConfig struct {
 	ScrapeProtocols []ScrapeProtocol `yaml:"scrape_protocols,omitempty"`
 	// Whether to scrape a classic histogram that is also exposed as a native histogram.
 	ScrapeClassicHistograms bool `yaml:"scrape_classic_histograms,omitempty"`
+	// Whether to convert a scraped classic histogram into a native histogram with custom buckets.
+	ConvertClassicHistograms bool `yaml:"convert_classic_histograms,omitempty"`
```

Comment:
You might want to add some parragraph about this in `docs/feature_flags.md` and update the configuration file `docs/configuration/configuration.md`.

Reply:
I'd rather not, I think at this point without WAL support #14730  this is way too experimental.

Reply:
This is not a feature flag, so it's fine, agree to put on official config docs later.

hmmm, given we will commit to this bool for the next 2y or so, should we double check if classic "use enum not bool" stereotype apply here? e.g. do we ever want to convert only some histograms e.g. by metric name (I would like to say "no"). Will we ever have option to convert to nh (with exponential or other schema buckets?)

Why not more explicit name? (no strong opinion)

```suggestion
	ConvertClassicHistogramsToNHCB bool `yaml:"convert_classic_histograms,omitempty"`
```

Reply:
Most of the times converting to exponential leads to a lot of error. Maybe we could detect if it's possible , but I'm not sure how much overhead it would be? @beorn7  , wdyt?

The variable isn't visible, so I assume you'd want to change the name of the config param as well if it stays a bool?

I think it's quite possible that people would like to control conversion on metric level. We already talked about maybe implementing max bucket/max schema control on metric level. I think it would have to be a special relabel kind to not reinvent the wheel.

Reply:
> The variable isn't visible, so I assume you'd want to change the name of the config param as well if it stays a bool?

ups, yes!

LGTM for relabel if that case will happen

Reply:
We discussed converting classic histograms to native histograms with an exponential schema quite a few times. IIRC the conclusion was consistently that the interpolation errors are just too uncontrollable to make this feasible. I think the same is true for any other future schema I can imagine (DDSketch, something log linear, whatever). The only thing that really works is NHCB, and if we really wanted to, we could even make NHCB work with other schemas in PromQL for the cases where the buckets happen to  coincide with the buckets in the other schema (e.g. you have a classic histogram with `le=1`, `le=2`, `le=4` etc. and no usage of the +inf bucket).
So I would be happy to settle with "we do not plan conversion of classic histograms to any other schemas".